src/main/java/core/basesyntax/strategy/handler/SupplyOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public SupplyOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer currentQuantity = fruitDao.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (purchaseQuantity < 0) {
+            throw new IllegalArgumentException("Invalid supply quantity - " + purchaseQuantity);
+        }
+        fruitDao.getStorage().put(transaction.getFruit(), currentQuantity + purchaseQuantity);
```

Comment:
Directly accessing the storage map with `fruitDao.getStorage().put()` breaks encapsulation. It's better to create a method in `FruitDao` that abstracts this operation, like `fruitDao.updateFruitQuantity(transaction.getFruit(), newQuantity);`

Reply:
I agree with this comment. If you have a DAO class, then don't allow direct manipulation with underlying data